Copy skip memory management options from mpfavorites to mplogin/mpdata

saagit opened the pull request:
The mpfavorites utility already has the -sme and -smx options to allow
the entry and exiting of memory management mode to be skipped.  This
change copies those options to the mplogin and mpdata utilities so
that multiple commands that require memory management mode can be
performed without having to go through the hassle of entering your PIN
for each command.

saagit committed:
Copy skip memory mgmnt enter/exit options from mpfavorites to mplogin & mpdata

The mpfavorites utility already has the -sme and -smx options to allow
the entry and exiting of memory management mode to be skipped.  This
change copies those options to the mplogin and mpdata utilities so
that multiple commands that require memory management mode can be
performed without having to go through the hassle of entering your PIN
for each command.

saagit committed:
Prompt user for password before opening Mooltipass

This change was prompted by trying to move a Mooltipass password to a
different context by using the Mooltipass to enter the password When
mplogin prompted "Enter the password to use:".  If you tried to do
this before this change, the message "HEY I GOT A 0xC4!" would be
printed by recv_packet() in mooltipass.py.  I don't see any reason for
mplogin to delay calling getpass.getpass() until after the Mooltipass
was opened so I moved it to be a part of the argparse handling in
mplogin's main_options() to avoid the 0xC4 problem.

saagit committed:
Fix various misspellings in comments and printed messages.

Simple changes that yield no functional change other than spellings in
printed messages.

saagit committed:
Remove out-of-date comment that "delete" needs implemented in mplogin.

saagit committed:
Remove obsolete Python constructs

The '<>' operator is considered obsolete in favor of the '!=' operator
per https://docs.python.org/2.7/reference/lexical_analysis.html and in
Python 3 it actually is an error.  There was a single instance of '<>'
in mooltipass_client.py which is changed to '!='.

The other change was to add parentheses around the handled exceptions
on 'except' statements in mpdata.py and mplogin.py.  Although not
officially obsolete, it is considered the old way to do things per
https://docs.python.org/2/tutorial/errors.html and also results in an
error in Python 3.

saagit committed:
Loosen the installation requirements

Two changes were made to setup.py and requirements.txt:
1) pycrypto is not used, so it was removed.
2) The current version of pyusb (1.0.2) works fine so the requirement
for pyusb was changed to be >= 1.0.0b2 instead of requiring exactly
1.0.0b2.

oSquat commented:
So I saw your first PR 043e139 come through before the holidays and forgot about it -- I apologize because now you've pushed a few other nice contributions behind something I don't want to accept.

As far as I recall, the sme/smx flags are for skipping entering/exiting management mode and this was useful for managing favorites because you might be adjusting many favorite slots in succession and otherwise would need to enter your pin each time to enter management mode. There's no need for management mode working with contexts or data, so having those flags doesn't make sense to me.

Do you know how to undo 043e139 leaving the other contributions? If you can do that, or if you start a new PR I'll accept all the other changes you've offered. Or maybe I can do this and I just don't know how (I'm far from being a git wizard). Let me know.

saagit commented:
Hi Vic,

No need to apologize at all.  I was surprised that when I pushed more 
things to head that they got incorporated into the initial pull request 
so that was a mistake on my part.  In the future, I plan to put each 
separate change into its own branch to make it easier to pick and choose 
changes.  I'll be happy to do that for this set of changes, but I would 
first like to explain why I find sme/smx useful for mplogin and mpdata 
to see if you might reconsider not wanting that change.

Initially, I didn't understand that the Mooltipass handled multiple 
usernames in a single context so nicely so I had created a number of 
entries in the form of:
   mplogin set 'site.com foo' -u='foo'
   mplogin set 'site.com bar' -u='bar'
Once I learned that the Mooltipass did handle multiple usernames per 
context so well, I wanted to rearrange things as if I had initially done 
things like:
   mplogin set 'site.com' -u='foo'
   mplogin set 'site.com' -u='bar'
To do that required a series of list/set/del commands to fix all the 
entries I had initially made.  mplogin (and mpdata) require memory 
management mode to be entered for both list and del.  It quickly became 
tedious to have to enter the PIN so often.  Imagine if you had to enter 
your password on your computer every time you wanted to do an "ls" or 
"rm" (or "dir" and "del" if you prefer).  Copying sme/smx from mpparams 
is a quick, easy and uniform way to avoid the tedium.

Although I personally haven't felt as compelling of a need for sme/smx 
on mpdata, the need is the same.  If someone wants to rename/rearrange 
the data contexts in the Mooltipass, it is inconvenient to have to enter 
the PIN for every list/del and the "cost" of solving the problem is very 
low.

I'm going to wait a day before I split the changes up into different 
branches which should make it easy to pick and choose which changes you 
want to pull.  If I've changed your mind and you have time to do the 
pull today, great, but if not it's no big deal and you'll be able to 
pick and choose the next day.

Thanks much,
   Scott

On 1/6/20 6:00 PM, Vic Simeone wrote:
> So I saw your first PR 043e139 
> <https://github.com/oSquat/mooltipy/commit/043e1390b9bf3f49091e92d7d9839f5ba46451be> 
> come through before the holidays and forgot about it -- I apologize 
> because now you've pushed a few other nice contributions behind 
> something I don't want to accept.
> 
> As far as I recall, the sme/smx flags are for skipping entering/exiting 
> management mode and this was useful for managing favorites because you 
> might be adjusting many favorite slots in succession and otherwise would 
> need to enter your pin each time to enter management mode. There's no 
> need for management mode working with contexts or data, so having those 
> flags doesn't make sense to me.
> 
> Do you know how to undo 043e139 
> <https://github.com/oSquat/mooltipy/commit/043e1390b9bf3f49091e92d7d9839f5ba46451be> 
> leaving the other contributions? If you can do that, or if you start a 
> new PR I'll accept all the other changes you've offered. Or maybe I can 
> do this and I just don't know how (I'm far from being a git wizard). Let 
> me know.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/oSquat/mooltipy/pull/15?email_source=notifications&email_token=ANGFYUEN6D2XN6643KQJFG3Q4PO2NA5CNFSM4J3O5CP2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEIHNQSQ#issuecomment-571398218>, 
> or unsubscribe 
> <https://github.com/notifications/unsubscribe-auth/ANGFYUEM3ZRUMQIBONDNQULQ4PO2NANCNFSM4J3O5CPQ>.
>

oSquat commented:
You are right! I'm embarrassed to now know my own software so well. I use the mooltipy weekly but don't often delete contexts or data. Ok, merged.

saagit commented:
No worries here and thanks much for the merge.

On 1/7/20 5:29 PM, Vic Simeone wrote:
> You are right! I'm embarrassed to now know my own software so well. I 
> use the mooltipy weekly but don't often delete contexts or data. Ok, merged.
